plugins/ccgx/fu-ccgx-dmc-device.c

```diff
@@ -0,0 +1,703 @@
+
+static gboolean
+fu_ccgx_dmc_device_send_sort_reset (FuCcgxDmcDevice *self,
+				    gboolean reset_later,
+				    GError **error)
+{
+	if (!g_usb_device_control_transfer (fu_usb_device_get_dev (FU_USB_DEVICE (self)),
+					    G_USB_DEVICE_DIRECTION_HOST_TO_DEVICE,
+					    G_USB_DEVICE_REQUEST_TYPE_VENDOR,
+					    G_USB_DEVICE_RECIPIENT_DEVICE,
+					    DMC_RQT_CODE_SOFT_RESET, /* request */
+					    reset_later, /* value */
+					    0, /* index */
+					    0, /* data */
+                                            0,  /* length */
+                                            NULL, /* actual length */
+					    DMC_CONTROL_TRANSFER_DEFAULT_TIMEOUT, /* timeout */
+					    NULL, error)) {
+		g_prefix_error (error, "send reset error:");
+		return FALSE;
+	}
+	return TRUE;
+}
+
+
+static gboolean
+fu_ccgx_dmc_device_send_start_upgrade (FuCcgxDmcDevice *self,
+				       guint8* custom_meta_data,
+				       guint16 custom_meta_size,
+				       GError **error)
+{
+	guint16 value = 0;
+	if (custom_meta_size > 0)
+		value = 1;
+	if (custom_meta_data == NULL && custom_meta_size > 0) {
+		g_set_error(error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "invalid  pointer at %s",__func__);
+			return FALSE;
+	}
+	if (!g_usb_device_control_transfer (fu_usb_device_get_dev (FU_USB_DEVICE (self)),
+					    G_USB_DEVICE_DIRECTION_HOST_TO_DEVICE,
+					    G_USB_DEVICE_REQUEST_TYPE_VENDOR,
+					    G_USB_DEVICE_RECIPIENT_DEVICE,
+					    DMC_RQT_CODE_UPGRADE_START, /* request */
+					    value, /* value */
+					    1, /* index, forced update for Adicora only, other dock will ignore it */
+					    custom_meta_data, /* data */
+					    custom_meta_size,  /* length */
+					    NULL, /* actual length */
+					    DMC_CONTROL_TRANSFER_DEFAULT_TIMEOUT, /* timeout */
+					    NULL, error)) {
+		g_prefix_error (error, "send reset error:");
+		return FALSE;
+	}
+	return TRUE;
+}
+
+static gboolean
+fu_ccgx_dmc_device_send_download_trigger (FuCcgxDmcDevice *self,
+					  DmcTriggerCode trigger,
+					  GError **error)
+{
+	if (!g_usb_device_control_transfer (fu_usb_device_get_dev (FU_USB_DEVICE (self)),
+					    G_USB_DEVICE_DIRECTION_HOST_TO_DEVICE,
+					    G_USB_DEVICE_REQUEST_TYPE_VENDOR,
+					    G_USB_DEVICE_RECIPIENT_DEVICE,
+					    DMC_RQT_CODE_TRIGGER, /* request */
+					    trigger, /* value */
+					    0, /* index */
+					    0, /* data */
+					    0,  /* length */
+					    NULL, /* actual length */
+					    DMC_CONTROL_TRANSFER_DEFAULT_TIMEOUT, /* timeout */
+					    NULL, error)) {
+		g_prefix_error (error, "send download trigger error:");
+		return FALSE;
+	}
+	return TRUE;
+}
+
+
+static gboolean
+fu_ccgx_dmc_device_send_fwct (FuCcgxDmcDevice *self,
+			      guint8* fwct_data,
+			      guint16 fwct_size,
+			      GError **error)
+{
+	if (fwct_data == NULL) {
+		g_set_error(error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "invalid  pointer at %s",__func__);
+			return FALSE;
```

Comment:
funky aligning

Reply:
Removed this code on today commit (ccgx: fix code alignment, modify code for review)